### PR TITLE
Add a few more ASR statement nodes

### PR DESCRIPTION
**Pull request by tapaswenipathak**
Ref https://github.com/lfortran/lfortran/issues/842.

### CODE REVIEW

File: doc/src/asr/asr_nodes/statement_nodes/explicitdeallocate.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+nonprocedure pointer targets. It also disassociates pointer. It is done by
+`Deallocate()` statement. If not allocated, it throws a runtime error.
+
+## Types
+
+Name of variable, must be a pointer or allocatable variable.
+
+## Examples
+
+```fortran
```
Comment (tapaswenipathak) [inline doc/src/asr/asr_nodes/statement_nodes/explicitdeallocate.md:33]: @certik can you add an example for `ExplicitDeallocate`?

File: doc/src/asr/asr_nodes/statement_nodes/implicitdeallocate.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+nonprocedure pointer targets. It also disassociates pointer. It is done by
+`Deallocate()` statement. If not allocated, it does nothing.
+
+## Types
+
+Name of variable, must be a pointer or allocatable variable.
+
+## Examples
+
+```fortran
```
Comment (tapaswenipathak) [inline doc/src/asr/asr_nodes/statement_nodes/implicitdeallocate.md:33]: @certik can you add an example for `ImplicitDeallocate`?

also, can the description have more information?

File: doc/src/asr/asr_nodes/statement_nodes/ifarithmetic.md [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+```fortran
+IfArithmetic(expr test, int lt_label, int eq_label, int gt_label)
+```
+
+### Arguments
+
+`test` contains conditional expression to be evaluated.
+`lt_label` contains
+`eq_label` contains
+`gt_label` contains
```
Comment (tapaswenipathak) [inline doc/src/asr/asr_nodes/statement_nodes/ifarithmetic.md:19]: @certik: what would these contain?

File: doc/src/asr/asr_nodes/statement_nodes/assert.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+program assert
+implicit none
+ASSERT(.true.)
+end program
+```
+
+ASR:
+
+```fortran
+
```
Comment (tapaswenipathak) [inline doc/src/asr/asr_nodes/statement_nodes/assert.md:47]: @certik: why does this error as?

Screenshot 2022-11-28 at 10 29 43 AM

File: doc/src/asr/asr_nodes/statement_nodes/where.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+     x = 0.0
+  end where
+  print *, x
+end program
+```
+
+ASR:
+
+```fortran
+```
```
Comment (tapaswenipathak) [inline doc/src/asr/asr_nodes/statement_nodes/where.md:55]: @certik: why do i get an error like:

Screenshot 2022-11-28 at 2 24 15 PM

**Comment by certik** [comment]
Thanks @tapaswenipathak for this! Yes, I'll document the rest and the feedback you gave us here. I don't have time right now, but we'll get to it.

CC @czgdp1807, @Smit-create, @Thirumalai-Shaktivel we can now keep documenting the ASR and fix / address the issues that Tapasweni raised above.

**Pull request merged**